js/src/Zone.js

```diff
@@ -1,67 +1,62 @@
 import ExpectedMark from "./ExpectedMark";
-import UXCapture from "./UXCapture";
+import UXBase from "./UXBase";
 
 /**
- * Zone represents collection of elements groupped together and corresponding phase of page load
+ * A `Zone` is a collection of elements on a page that corresponds
+ * to a given phase of page load. (i.e. all elements in `ux-destination-verfied`)
  *
- * Only one set of zones can be tracked for the same view at one point in time
+ * Example props:
+ *
+ * {
+ *  name: "ux-destination-verified",
+ *  marks: ["ux-image-online-logo", "ux-image-inline-logo"]
+ *  onMeasure: measureName => {}
+ *  onMark: markName => {}
+ * }
  */
-export default class Zone {
-  // constructs individual zone object
-  constructor(config) {
-    // {
-    //   name: "ux-destination-verified",
-    //   marks: ["ux-image-online-logo", "ux-image-inline-logo"]
-    //   selectors: ["img.logo"],
-    //   startMark: "navigationStart",
-    //   onMeasure: measureName => {}
-    //   onMark: markName => {}
-    // }
-
-    // name to be used for UserTiming measures
-    this.measureName = config.name;
+export default class Zone extends UXBase {
+  /**
+   * Creates a Zone
+   * @param {object} props
+   */
+  constructor(props) {
+    super(props);
 
-    // handling deprecated "label" keys in backwards-compatible way
-    if (config.label) {
+    // Handle deprecated "label" keys
+    if (this.props.label) {
       console.warn(
         "[ux-capture] Deprecation Warning: `label` keys on configuration object were renamed to `name` as of verision v2.0.0",
         "Will be removed in v3.0.0"
       );
-
-      if (!config.name) {
-        this.measureName = config.label;
-      }
     }
+  }
 
-    // callback to execute when Zone is complete
-    this.onMeasure = config.onMeasure;
-
-    // callback for marks to call when they are complete
-    this.onMark = config.onMark;
+  // Name used for UserTiming measures
+  measureName = this.props.name || this.props.label;
 
-    this.startMarkName = config.startMarkName || "navigationStart";
+  // Callback to execute when Zone is complete (all marks recorded)
+  onMeasure = this.props.onMeasure;
 
-    // look up existing mark object or create a new one
+  // Callback for marks to call when they are complete (recorded)
+  onMark = this.props.onMark;
```

Comment:
can we remove the `this.onMark` and `this.onMeasure` aliases?

Reply:
yep, removed